### PR TITLE
[MSFT] Rename DeviceDB to PlacementDB

### PR DESCRIPTION
**Pull request by jptech**
No functional changes. This PR simply renames DeviceDB to PlacementDB within the MSFT dialect. The "DeviceDB" name may be used in a future PR and live alongside the PlacementDB.

### CODE REVIEW

**Review by teqdruid — APPROVED** [review]
Thanks Parker! You've got some clang-format issues, but one you work those out you're good to go. I'd recommend installing a clang-format based autoformatter in your editor. I have one installed in vscode and have it set to format the changed lines every time I save.

The python format "error" is bogus... Been meaning to fix that.

**Comment by mikeurbach** [comment]
> The python format "error" is bogus... Been meaning to fix that.

Aw crap, this is my nemesis. Issue filed here a while back: https://github.com/llvm/circt/issues/1162.

File: include/circt/Dialect/MSFT/PlacementDB.h [hunk 1]
```diff
@@ -1,4 +1,5 @@
-//===- PlacementDB.h - Device database -----------------------------*- C++ -*-===//
+//===- PlacementDB.h - Device database -----------------------------*- C++
```
Comment (teqdruid) [inline include/circt/Dialect/MSFT/PlacementDB.h:1]: This doesn't look right...

Comment (jptech) [reply include/circt/Dialect/MSFT/PlacementDB.h:1]: Good catch. I also somehow missed updating the include guard name. Fixing...

File: lib/Dialect/MSFT/PlacementDB.cpp [hunk 2]
```diff
@@ -1,12 +1,13 @@
-//===- DeviceDB.cpp - Implement a device database -------------------------===//
+//===- PlacementDB.cpp - Implement a device database
+//-------------------------===//
```
Comment (teqdruid) [inline lib/Dialect/MSFT/PlacementDB.cpp:2]: This also doesn't look quite right

**Comment by teqdruid** [comment]
Thanks Parker!

**Pull request merged**